### PR TITLE
Improve vanilla test coverage

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

File: javagen/src/main/java/com/azure/autorest/mapper/ModelPropertyMapper.java [hunk 1]
```diff
@@ -105,8 +105,8 @@ public ClientModelProperty map(Property property) {
                 && sequence.getElementType().getSerialization().getXml() != null
                 && sequence.getElementType().getSerialization().getXml().getName() != null) {
                 builder.xmlListElementName(sequence.getElementType().getSerialization().getXml().getName());
-            } else if(sequence.getSerialization() != null && sequence.getSerialization().getXml() != null){
-                builder.xmlListElementName(sequence.getSerialization().getXml().getName());
+            } else {
+                builder.xmlListElementName(sequence.getElementType().getLanguage().getDefault().getName());
```
Comment (jianghaolu) [inline javagen/src/main/java/com/azure/autorest/mapper/ModelPropertyMapper.java:109]: @srnagar if serialization.xml.name isn't available - it doesn't seem right to use the array schema's xml name for the item name. For now I'm using the name of the item as a fallback and seems to generate this correctly so far.

Comment (srnagar) [reply javagen/src/main/java/com/azure/autorest/mapper/ModelPropertyMapper.java:109]: Sounds good.

File: vanilla-tests/pom.xml [hunk 2]
```diff
@@ -18,12 +18,12 @@
...
       <groupId>com.azure</groupId>
       <artifactId>azure-core</artifactId>
-      <version>1.2.0</version>
+      <version>1.4.0</version>
     </dependency>
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-core-http-netty</artifactId>
-      <version>1.1.0</version>
+      <version>1.4.0</version>
```
Comment (srnagar) [inline vanilla-tests/pom.xml:26]: Just using azure-core-http-netty as dependency would transitively bring in azure-core as well. Also, we have 1.5.0 as the latest netty version.

Comment (weidongxu-microsoft) [reply vanilla-tests/pom.xml:26]: Ideally, netty should be in test scope.

**Pull request merged**